### PR TITLE
Fix expression snapping above staff

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #18498

### CODE REVIEW

File: src/engraving/layout/v0/tlayout.cpp [hunk 1]
```diff
@@ -2693,8 +2693,10 @@ void TLayout::layout(HairpinSegment* item, LayoutContext& ctx)
                 }
                 if (ed->ipos().y() != ny) {
                     ed->setPosY(ny);
-                    if (ed->snappedExpression()) {
-                        ed->snappedExpression()->setPosY(ny);
+                    Expression* snappedExpression = ed->snappedExpression();
+                    if (snappedExpression) {
+                        double yOffsetDiff = snappedExpression->offset().y() - ed->offset().y();
+                        ed->snappedExpression()->setPosY(ny - yOffsetDiff);
```
Comment (RomanPudashkin) [inline src/engraving/layout/v0/tlayout.cpp:2699]: Please use snappedExpression->setPosY(...) instead of ed->snappedExpression()->setPosY(...)

Comment (mike-spa) [reply src/engraving/layout/v0/tlayout.cpp]: oops! done

**Review by RomanPudashkin — APPROVED** [review]

**Review by oktophonie — APPROVED** [review]

File: src/engraving/style/styledef.cpp [hunk 2]
```diff
@@ -921,6 +921,8 @@ const std::array<StyleDef::StyleValue, size_t(Sid::STYLES)> StyleDef::styleValue
     { Sid::expressionAlign,               "expressionAlign",              Align(AlignH::LEFT, AlignV::BASELINE) },
     { Sid::expressionPlacement,           "expressionPlacement",          PlacementV::BELOW },
     { Sid::expressionOffset,              "expressionOffset",             PointF(.0, 2.5) },
+    { Sid::expressionPosAbove,            "expressionPosAbove",           PointF(.0, -1.5) },
+    { Sid::expressionPosAbove,            "expressionPosAbove",           PointF(.0, 2.5) },
```
Comment (bakajikara) [inline src/engraving/style/styledef.cpp:925]: Shouldn't this be expressionPosBelow?

Comment (mike-spa) [reply src/engraving/style/styledef.cpp:925]: urghhhh mistake, thank for spotting it 👍

**Pull request merged**